object-filter/src/main/java/org/infinispan/objectfilter/impl/predicateindex/ProtobufMatcherEvalContext.java

```diff
@@ -32,7 +33,12 @@ public ProtobufMatcherEvalContext(Object userContext, Object eventType, Object i
       super(userContext, eventType, instance);
       this.serializationContext = serializationContext;
       try {
-         ProtobufParser.INSTANCE.parse(this, wrappedMessageDescriptor, (byte[]) getInstance());
+         Object inst = getInstance();
+         if (inst instanceof WrappedByteArray) {
```

Comment:
since you know the classes, maybe inst.getClass()==WrappedByteArray.class (or even inst.getClass().isArray() would be even cheaper)